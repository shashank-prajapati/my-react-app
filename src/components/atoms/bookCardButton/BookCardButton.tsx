import { Button, ButtonProps } from "@mui/material";
import React from "react";
import ButtonAtom from "../button/ButtonAtom";
interface Props {
  btnText?: string;
  isHovered?: Boolean;
  startIcon?: React.ReactNode
}
const BookCardButton = (props: Props) => {
  const { btnText, startIcon, isHovered} = props;
  let backgroundColor = "white";
  let textColor = "secondary.main";
  if (isHovered) {
    backgroundColor = "secondary.main";
    textColor = "white";
  }
  return (
    <div>
      <ButtonAtom
        variant="contained"
        width="100%"
        textColor={textColor}
        backgroundColor={backgroundColor}
        hoverTextColor="white"
        hoverBackGroundColor="secondary.main"
        startIcon={startIcon}
        size="large"
      >
        {btnText}
      </ButtonAtom>
    </div>
  );
};
export default BookCardButton;
