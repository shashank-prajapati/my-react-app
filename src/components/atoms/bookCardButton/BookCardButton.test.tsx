import * as React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import BookCardButton from "./BookCardButton";

describe("Testing Book Card Button", () => {
  it("should match the default snapshot", () => {
    const { container } = render(<BookCardButton btnText="Add to library" />);
    expect(container).toMatchSnapshot();
  });

  it("should match the dhover snapshot", () => {
    const { container } = render(
      <BookCardButton btnText="Add to library" isHovered={true} />
    );
    expect(container).toMatchSnapshot();
  });
});
