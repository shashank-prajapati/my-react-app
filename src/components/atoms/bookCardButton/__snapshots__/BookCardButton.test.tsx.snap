// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Book Card Button should match the default snapshot 1`] = `
<div>
  <div>
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jbktht-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Add to library
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Testing Book Card Button should match the dhover snapshot 1`] = `
<div>
  <div>
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-47f4e9-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Add to library
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
